odl/space/space_utils.py

```diff
@@ -281,6 +284,303 @@ def rn(shape, dtype=None, impl='numpy', **kwargs):
     return rn
 
 
+class auto_weighting(OptionalArgDecorator):
+
+    """Make an unweighted adjoint automatically account for weightings.
+
+    This decorator can be used in `Operator` subclasses to automatically
+    perform weighting of the adjoint operator based on the space
+    weightings. It is used as ::
+
+        class MyOperator(odl.Operator):
+
+            @auto_weighting
+            @property
+            def adjoint(self):
+                return MyOperatorUnweightedAdjoint(...)
+
+    In this context, the term "unweighted adjoint" of an operator ``A``
+    refers to the the adjoint of ``A`` defined between **unweighted
+    Euclidean spaces**. In particular, discretized ``L^p`` type spaces are
+    interpreted as weighted variants of ``R^n`` or ``C^n`` spaces.
+
+    For instance, an (m x n) real matrix ``M`` is also an operator
+    ``M: R^n -> R^m``, and its unweighted adjoint is defined by the
+    transposed matrix ``M^T: R^m -> R^n``.
+
+    Similarly, when an integration operator ``A: L^2(I) -> R`` on an
```

Comment:
Wouldnt always using the R^n -> R style be better?

Reply:
What do you mean by "using"? In this example?